### ISSUE TITLE
Build after clippy --fix fails

### ISSUE THREAD

**Issue by aviv-augustinjuricic** [rust-lang/rust-clippy#14502]
### Summary

Build after clippy --fix fails:

```
D:\Code\rs-piebin [bug/iced-12-not-working]> cargo clippy --fix --allow-dirty
    Checking piebin v0.1.2 (D:\Code\rs-piebin)
warning: failed to automatically apply fixes suggested by rustc to crate `piebin`

after fixes were automatically applied the compiler reported errors within these files:

  * src\filesystem\plain.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0106]: missing lifetime specifier
   --> src\filesystem\plain.rs:111:53
    |
111 |     fn set_size(self: &mut NodeRef, size: usize) -> &mut NodeRef {
    |                       ------------                  ^ expected named lifetime parameter
    |
    = help: this function's return type contains a borrowed value, but the signature does not say which `self` it is borrowed from
help: consider introducing a named lifetime parameter
    |
111 |     fn set_size<'a>(self: &'a mut NodeRef, size: usize) -> &'a mut NodeRef {
    |                ++++        ++                               ++

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0106`.
Original diagnostics will follow.

warning: the following explicit lifetimes could be elided: 'a
   --> src\filesystem\mt_double_link.rs:150:17
    |
150 |     fn set_size<'a>(self: &'a mut NodeRef, size: usize) -> &'a mut NodeRef {
    |                 ^^         ^^                               ^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
    = note: `#[warn(clippy::needless_lifetimes)]` on by default
help: elide the lifetimes
    |
150 -     fn set_size<'a>(self: &'a mut NodeRef, size: usize) -> &'a mut NodeRef {
150 +     fn set_size(self: &mut NodeRef, size: usize) -> &mut NodeRef {
    |

warning: the following explicit lifetimes could be elided: 'a
   --> src\filesystem\mt_single_link.rs:140:17
    |
140 |     fn set_size<'a>(self: &'a mut NodeRef, size: usize) -> &'a mut NodeRef {
    |                 ^^         ^^                               ^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
help: elide the lifetimes
    |
140 -     fn set_size<'a>(self: &'a mut NodeRef, size: usize) -> &'a mut NodeRef {
140 +     fn set_size(self: &mut NodeRef, size: usize) -> &mut NodeRef {
    |

warning: the following explicit lifetimes could be elided: 'a
   --> src\filesystem\plain.rs:111:17
    |
111 |     fn set_size<'a>(self: &'a mut NodeRef, size: usize) -> &'a mut NodeRef {
    |                 ^^         ^^                               ^^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes
help: elide the lifetimes
    |
111 -     fn set_size<'a>(self: &'a mut NodeRef, size: usize) -> &'a mut NodeRef {
111 +     fn set_size(self: &mut NodeRef, size: usize) -> &mut NodeRef {
    |

warning: `piebin` (bin "piebin") generated 3 warnings (run `cargo clippy --fix --bin "piebin"` to apply 3 suggestions)
warning: failed to automatically apply fixes suggested by rustc to crate `piebin`

after fixes were automatically applied the compiler reported errors within these files:

  * src\filesystem\plain.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0106]: missing lifetime specifier
   --> src\filesystem\plain.rs:111:53
    |
111 |     fn set_size(self: &mut NodeRef, size: usize) -> &mut NodeRef {
    |                       ------------                  ^ expected named lifetime parameter
    |
    = help: this function's return type contains a borrowed value, but the signature does not say which `self` it is borrowed from
help: consider introducing a named lifetime parameter
    |
111 |     fn set_size<'a>(self: &'a mut NodeRef, size: usize) -> &'a mut NodeRef {
    |                ++++        ++                               ++

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0106`.
Original diagnostics will follow.

warning: `piebin` (bin "piebin" test) generated 3 warnings (3 duplicates)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.98s
```

### Reproducer

I tried this code:

```rust
<code>
```

I expected to see this happen:

Instead, this happened:

### Version

```text

```

**Comment by samueltardieu** [comment]
Without code to reproduce it, it is hard to determine whether this is a Clippy bug or not. It might also be related to https://github.com/rust-lang/rust/issues/135659.

**Comment by aviv-augustinjuricic** [comment]
@samueltardieu Trying to extract code to reproduce it, but it is proving difficult. For some reason, it works when the codebase is smaller.

**Comment by lf-** [comment]
I believe that I hit this when running clippy fix on https://github.com/facebook/buck2 fyi. So if you want a test case, you can probably minimize one from there.
